x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlAssignmentNotifier.java

```diff
@@ -136,48 +144,71 @@ private void auditMlTasks(
                             msg += "] on node [" + nodeName + "]";
                         }
                         anomalyDetectionAuditor.info(jobId, msg);
-                    } else if (anomalyDetectionAuditor.includeNodeInfo()) {
+                    } else {
                         if (alwaysAuditUnassigned) {
-                            anomalyDetectionAuditor.warning(
-                                jobId,
-                                "No node found to start datafeed ["
-                                    + datafeedParams.getDatafeedId()
-                                    + "]. Reasons ["
-                                    + currentAssignment.getExplanation()
-                                    + "]"
-                            );
+                            if (anomalyDetectionAuditor.includeNodeInfo()) {
+                                anomalyDetectionAuditor.warning(
+                                    jobId,
+                                    "No node found to start datafeed ["
+                                        + datafeedParams.getDatafeedId()
+                                        + "]. Reasons ["
+                                        + currentAssignment.getExplanation()
+                                        + "]"
+                                );
+                            } else {
+                                anomalyDetectionAuditor.warning(jobId, "Awaiting capacity to start datafeed.");
```

Comment:
The datafeed can be named differently to the job, so please include the datafeed ID here, for example:

```
"Awaiting capacity to start datafeed [" + datafeedParams.getDatafeedId() + "]."
```